src/main/java/frc/robot/subsystems/gripper/Gripper.java

```diff
@@ -0,0 +1,81 @@
+package frc.robot.subsystems.gripper;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Gripper extends SubsystemBase {
+    private static Gripper INSTANCE;
+    private final GripperIO io;
+    private final GripperInputsAutoLogged inputs = GripperIO.inputs;
+
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(0, 0);
+    @AutoLogOutput private Pose3d pose3d = new Pose3d(0, 0, 0, new Rotation3d());
+
+    private final MechanismRoot2d root = mechanism2d.getRoot("Gripper", 0, 0);
+    private final MechanismLigament2d gripperLigament =
+            root.append(new MechanismLigament2d("Gripper", 0, 0));
+
+    private Gripper(GripperIO io) {
+        this.io = io;
+    }
+
+    public static Gripper getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(GripperIO io) {
+        INSTANCE = new Gripper(io);
+    }
+
+    public boolean hasNote() {
+        return inputs.hasNote;
+    }
+
+    public boolean atSetpoint() {
+        return inputs.currentAngle.isNear(inputs.angleSetpoint, 0.02);
```

Comment:
0.02 should be stored it a constant of type `Measure<Dimensionless>`.

Reply:
0eb4bbbf1420f1f9b517fd4b1042369b6ed39f93

Reply:
This is a constant, it shouldn't be mutable.

Reply:
c7f81f291ce9f5b71be3890a6fdcb9fd61be6150